JesusFilm/core
# refactor: journeyDuplicate query

# Description

### Issue

Components needed for the resources hub need refactored out into the shared space to be used by both journeys-admin and watch. 

We need to refactor out useJourneyDuplicate query because it is needed by CreateJourneyButton which will be refactored out in a future PR.

[Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/37663178/todos/7452920587)

### Solution

This PR refactors the useJourneyDuplicate query into the journeys/ui/ space. If I were able to dynamically import this useJourneyDuplicate query into the CreateJourneyButton I would, but I don't think that can be done. The component logic won't let this query get called unless the user is logged in.

# External Changes

n/a

# Additional information

Big-bang discovery branch is here https://github.com/JesusFilm/core/pull/2810.

## How To Test
There should be no behavioural impact so ensure application still performs as expected. These components are currently only being used in journeys-admin so regression test as appropriate.
One should be able to use the useJourneyDuplicate query in another project.
